crates/re_space_view/src/data_query_blueprint.rs

```diff
@@ -0,0 +1,401 @@
+        overrides: EntityPropertyMap,
+    }
+
+    impl PropertyResolver for StaticPropertyResolver {
+        fn resolve_entity_overrides(&self, _ctx: &StoreContext<'_>) -> EntityPropertyMap {
+            self.overrides.clone()
+        }
+
+        fn resolve_root_override(&self, _ctx: &StoreContext<'_>) -> EntityProperties {
+            EntityProperties::default()
+        }
+    }
+
+    #[test]
+    fn test_query_results() {
+        let mut recording = StoreDb::new(StoreId::random(re_log_types::StoreKind::Recording));
+        let blueprint = StoreDb::new(StoreId::random(re_log_types::StoreKind::Blueprint));
+
+        let overrides = EntityPropertyMap::default();
+        let resolver = StaticPropertyResolver { overrides };
+
+        let timeline_frame = Timeline::new_sequence("frame");
+        let timepoint = TimePoint::from_iter([(timeline_frame, 10.into())]);
+
+        // Set up a store DB with some entities
+        for entity_path in ["parent", "parent/skipped/child1", "parent/skipped/child2"] {
+            let row_id = RowId::random();
+            let point = MyPoint::new(1.0, 2.0);
+            let row = DataRow::from_component_batches(
+                row_id,
+                timepoint.clone(),
+                entity_path.into(),
+                [&[point] as _],
+            )
+            .unwrap();
+
+            recording.add_data_row(row).unwrap();
+        }
+
+        let ctx = StoreContext {
+            blueprint: &blueprint,
+            recording: Some(&recording),
+            all_recordings: vec![],
+        };
+
+        let scenarios: Vec<(Vec<&str>, Vec<&str>)> = vec![
+            (
+                vec!["/"],
+                vec![
+                    "/",
+                    "parent/",
+                    "parent",
+                    "parent/skipped/", // Not an exact match and not found in tree
+                    "parent/skipped/child1", // Only child 1 has ViewParts
+                ],
+            ),
+            (
+                vec!["parent/skipped/"],
+                vec![
+                    "/",
+                    "parent/",               // Only included because is a prefix
+                    "parent/skipped/",       // Not an exact match and not found in tree
+                    "parent/skipped/child1", // Only child 1 has ViewParts
+                ],
+            ),
+            (
+                vec!["parent/skipped", "parent/skipped/child2", "parent/"],
+                vec![
+                    "/",
+                    "parent/",
+                    "parent",
+                    "parent/skipped/",
+                    "parent/skipped",        // Included because an exact match
+                    "parent/skipped/child1", // Included because an exact match
+                    "parent/skipped/child2",
+                ],
+            ),
+        ];
+
+        let mut entities_per_system_per_class = EntitiesPerSystemPerClass::default();
```

Comment:
would have been less confusing to read if this thing was at the top - was starting to read the scenarios and then got confused about what determines the ViewSystems